src/main/java/org/wildfly/security/auth/provider/FileSystemSecurityRealm.java

```diff
@@ -245,30 +243,23 @@ public String getName() {
             return name;
         }
 
-        public CredentialSupport getCredentialSupport(final Class<?> credentialType, final String algorithmName) throws RealmUnavailableException {
-            List<Object> credentials = loadCredentials();
-            for (Object credential : credentials) {
-                if (credentialType.isInstance(credential)) {
-                    return CredentialSupport.FULLY_SUPPORTED;
-                }
-            }
-            return CredentialSupport.UNSUPPORTED;
+        public CredentialSupport getCredentialSupport(final String credentialName) throws RealmUnavailableException {
+            Map<String, Object> credentials = loadCredentials();
+            return credentials.containsKey(credentialName) ? CredentialSupport.FULLY_SUPPORTED : CredentialSupport.UNSUPPORTED;
         }
 
-        public <C> C getCredential(final Class<C> credentialType, final String algorithmName) throws RealmUnavailableException {
-            List<Object> credentials = loadCredentials();
-            for (Object credential : credentials) {
-                if (credentialType.isInstance(credential)) {
-                    return credentialType.cast(credential);
-                }
+        public <C> C getCredential(final String credentialName, final Class<C> credentialType) throws RealmUnavailableException {
```

Comment:
Even though this method is null-safe in implementation, the parameters should be null-checked for correctness.  This is true of the other methods as well.